### ISSUE TITLE
Rethink / Document the tantivy index format.

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-oss/tantivy#981]

**Comment by drusellers** [comment]
@fulmicoton since there is a pretty big rethink open, have you looked at how Vespa does things? I've been reading a lot about that engine with all of the ES circus going on.

**Comment by fulmicoton** [comment]
@drusellers Yes I did have a look at Vespa. Mostly for the way they distribute things, but not their index format.

**Comment by drusellers** [comment]
https://www.youtube.com/watch?v=SzZ_A9G6PMY

Checkout 41:00 - where they talk about read/write workloads

59:00 - a discussion about in place updates

**Comment by ppodolsky** [comment]
It is further development of BMW algorithm. I just leave it here, think there is no any other better place for it. https://doi.org/10.1145/3077136.3080780

I'm gonna dive into in a while, had a look but although cannot say yet if it is applicable to Tantivy or is just another one pure academical research.

**Comment by fulmicoton** [comment]
Assigning to @fmassot , but he will just lead the effort.  People who want to join the discussion is welcome.
